junit-platform-engine/src/main/java/org/junit/platform/engine/TestEngine.java

```diff
@@ -192,4 +193,13 @@
 		return Optional.of(PackageUtils.getAttribute(getClass(), Package::getImplementationVersion).orElse(fallback));
 	}
 
+	/**
+	 * Print help message in "help style".
+	 *
+	 * @since 1.5
+	 */
+	default void printHelpMessage(PrintWriter writer) {
+		writer.printf("# Engine with ID '%s' doesn't provide any configuration help.%n", getId());
+		writer.println();
+	}
```

Comment:
Instead of printing a help message we should provide a richer model that can be consumed by console tools and IDEs, e.g. a `AvailableConfigurationParameters` object that contains a list of `AvailableConfigurationParameter` objects, each with a name, default value and description.

Reply:
Agreed. Will come up with a second spike next week.